api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -179,339 +179,355 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 		namespace = "default"
 	}
 
-	mLog.DebugfCtx(ctx, "  P (Materialize Processor): masterialize %v in namespace %s", prefixedNames, namespace)
+	mLog.DebugfCtx(ctx, "  P (Materialize Processor): materialize %v in namespace %s", prefixedNames, namespace)
 
-	var catalogs []model.CatalogState
-	catalogs, err = i.ApiClient.GetCatalogs(ctx, namespace, i.Config.User, i.Config.Password)
-
-	if err != nil {
-		mLog.ErrorfCtx(ctx, "Failed to get catalogs: %s", err.Error())
+	catalogs := make(map[string]model.CatalogState)
+	errorMessage := "Failed to get all catalogs: "
+	for _, object := range prefixedNames {
+		object := api_utils.ConvertReferenceToObjectName(object)
+		catalog, err := i.ApiClient.GetCatalog(ctx, object, namespace, i.Config.User, i.Config.Password)
+		if err != nil {
+			errorMessage = fmt.Sprintf("%s %s(reason: %s).", errorMessage, object, err.Error())
+			continue
+		}
+		catalogs[object] = catalog
+	}
+	if len(catalogs) < len(prefixedNames) {
+		mLog.ErrorCtx(ctx, errorMessage)
 		providerOperationMetrics.ProviderOperationErrors(
 			materialize,
 			functionName,
 			metrics.ProcessOperation,
 			metrics.RunOperationType,
 			v1alpha2.CatalogsGetFailed.String(),
 		)
-		return outputs, false, err
+		return outputs, false, v1alpha2.NewCOAError(nil, errorMessage, v1alpha2.NotFound)
```

Comment:
I am not sure if NotFound is appropriate for all errors. Sometimes getCatalog call can also failed with transient issue like service unavailability.

Reply:
How about `v1alpha2.CatalogsGetFailed`?

Reply:
I would prefer to return badrequest for NotFound errors so that we won't retry and return InternalError state for all others.